server/src/main/java/org/opensearch/extensions/ExtensionDependencyResponse.java

```diff
@@ -0,0 +1,70 @@
+/*
+* Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.extensions;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.transport.TransportResponse;
+
+/**
+ * The response for getting the Extension Dependency.
+ *
+ * @opensearch.internal
+ */
+public class ExtensionDependencyResponse extends TransportResponse {
+    private List<DiscoveryExtensionNode> dependency;
```

Comment:
(Style/Naming) Since this is a list of possibly more than one dependency, would prefer to use the plural `dependencies` here (and the getter, toString, equals, hashCode, etc.).